### PR TITLE
Adding r-rlemon interface to C++ LEMON graph library

### PR DESCRIPTION
**Pull request by cpalmer718**
This recipe for the [rlemon package from CRAN](https://cran.r-project.org/web/packages/rlemon/index.html) was generated with [conda_r_skeleton_helper](https://github.com/bgruening/conda_r_skeleton_helper).

Notes:
- This library is intended as a future dependency for the CRAN package optmatch, but following the instructions in the skeleton helper, this PR is just for the single package.
- This library is not to be confused with [r-lemon](https://anaconda.org/conda-forge/r-lemon), which is remarkably something else. The unfortunate partial name collision is inherited from CRAN and is seemingly unavoidable.
- The recipe succeeds with `build-locally.py` on a linux-64 machine.
- I'm the only listed individual maintainer, and agree to being there.

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Review by mfansler — CHANGES_REQUESTED** [review]
Everything looks good, except need the `missing_dso_whitelist` rather than changing `rpaths`.

**Comment by cpalmer718** [comment]
@mfansler Thanks so much for the comment, that was super helpful. Windows builds are pretty far from my expertise, so that would have taken me way longer to find!

File: recipes/r-rlemon/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    - lib/
+  missing_dso_whitelist:
+    - '*/R.dll'  # [win]
+requirements:
+  build:
+    - {{ compiler('c') }}              # [not win]
+    - {{ compiler('m2w64_c') }}        # [win]
+    - {{ compiler('cxx') }}            # [not win]
+    - {{ compiler('m2w64_cxx') }}      # [win]
+    - m2w64-libwinpthread-git          # [win]
```
Comment (mfansler) [inline recipes/r-rlemon/meta.yaml:29]: This shouldn't be necessary.

```suggestion
```

File: recipes/r-rlemon/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    - {{ posix }}sed               # [win]
+    - {{ posix }}coreutils         # [win]
+    - {{ posix }}zip               # [win]
+    - cross-r-base {{ r_base }}    # [build_platform != target_platform]
+  host:
+    - r-base
+    - r-rcpp >=1.0.5
+  run:
+    - r-base
+    - m2w64-libwinpthread-git      # [win]
```
Comment (mfansler) [inline recipes/r-rlemon/meta.yaml:41]: Use the jinja variable:
```suggestion
    - {{ native }}libwinpthread-git  # [win]

**Review by mfansler — CHANGES_REQUESTED** [review]

**Review by mfansler — APPROVED** [review]
@conda-forge/help-r looks good to merge!

**Comment by mfansler** [comment]
@conda-forge/help-r please review

**Pull request merged**